website_sale_recently_viewed_products/models/product_view.py

```diff
@@ -0,0 +1,47 @@
+from openerp import api, fields, models, _
+
+
+class ProductHistory(models.Model):
+    _name = 'website.sale.product.view'
+    _description = 'Ecommerce Product Views'
+
+    sessionid = fields.Char('Session ID', index=True)
+    product_id = fields.Many2one('product.template', 'Product')
+    last_view_datetime = fields.Datetime(
+        'Last view datetime', default=fields.Datetime.now)
+
+    _sql_constraints = [
+        ('unique_session_product', 'UNIQUE(sessionid, product_id)',
+         'There is already a record for this product and session')
+    ]
+    _order = 'last_view_datetime DESC'
+
+    @api.multi
+    def human_readable_datetime_difference(self, now=None):
+        """
+        Return an human readable form of the difference between the supplied
+        datetime (or the current datetime if not supplied) and the history
+        record ``last_view_datetime``.
+        """
+        if now is None:
+            now = fields.Datetime.from_string(fields.Datetime.now())
+        timedifference = now - fields.Datetime.from_string(
+            self.last_view_datetime)
+        minutes = timedifference.seconds / 60
+        hours = timedifference.seconds / 3600
```

Comment:
I recommend using the `//` operator for integer division as some day we may get python3 support in Odoo...